crates/sargon-uniffi/src/system/sargon_os/entities_controlled_by_factor_source/support/entities_controlled_by_factor_source.rs

```diff
@@ -0,0 +1,27 @@
+use crate::prelude::*;
+use sargon::EntitiesControlledByFactorSource as InternalEntitiesControlledByFactorSource;
+
+/// This is the result of checking what entities are controlled by a given `FactorSource`.
+#[derive(Clone, PartialEq, Eq, InternalConversion, uniffi::Record)]
+pub struct EntitiesControlledByFactorSource {
+    /// The factor source that controls the entities.
+    pub factor_source: FactorSource,
+
+    /// Whether the mnemonic of the factor source is present in keychain.
+    pub is_mnemonic_present_in_keychain: bool,
```

Comment:
many of the fields here are `DeviceFactorSource` specific. If the WHOLE query is that, we should name it `EntitiesUsingDeviceFactorSource`? Or we should generalize it, probably with enums with associated data, so that `is_mnemonic_present_in_keychain: bool` only exists when the factor source is of kind `device`